modules/apps/commerce/commerce-channel-web/src/main/java/com/liferay/commerce/channel/web/internal/display/context/CommerceChannelDisplayContext.java

```diff
@@ -413,6 +432,17 @@ public boolean hasAddChannelPermission() {
 			CPActionKeys.ADD_COMMERCE_CHANNEL);
 	}
 
+	public boolean hasManageLinkSupplierPermission() {
+		ThemeDisplay themeDisplay =
+			(ThemeDisplay)httpServletRequest.getAttribute(
+				WebKeys.THEME_DISPLAY);
+
+		PermissionChecker permissionChecker =
+			themeDisplay.getPermissionChecker();
+
+		return permissionChecker.isCompanyAdmin();
```

Comment:
You are saying the only the company-admin can update it, but this should be linked to the manage-channel or update channel permission instead, or if you really want to be specific a new permission action UPDATE_SUPPLIER_LINK (or something like that)

Reply:
We are now checking the VIEW_COMMERCE_CHANNELS permission to enable/disable the field